src/Core/Utility/JsonEx.re

```diff
@@ -53,6 +53,31 @@ let getKeys = json => {
   loop("", json);
 };
 
+let parseYojsonErrorMessage = {
+  open Oniguruma;
+  let yojsonRegExp =
+    OnigRegExp.create("Line ([0-9]+), bytes ([0-9]+)-([0-9]+):\n(.*)$");
+
+  msg => {
+    yojsonRegExp
+    |> Result.to_option
+    |> OptionEx.flatMap(regex => {
+         let matches = regex |> OnigRegExp.search(msg, 0);
+         if (Array.length(matches) < 5) {
+           None;
+         } else {
+           let line = OnigRegExp.Match.getText(matches[1]) |> int_of_string;
+           let startByte =
+             OnigRegExp.Match.getText(matches[2]) |> int_of_string;
+           let endByte =
+             OnigRegExp.Match.getText(matches[3]) |> int_of_string;
+           let message = OnigRegExp.Match.getText(matches[4]);
+           Some((line, startByte, endByte, message));
```

Comment:
This isn't a very usable return value. You could use a `Location.t` to improve it a bit, but why not move it to `Core.Json` and do it properly with a record? Not sure this belongs in `Utility` in the first place, as it seems to fit in well there.

Reply:
I moved it to `Core.Json` as an `Error.t = { range: Range.t, message: string }` - it is more ergonomic to use.